tests/pytests/unit/utils/test_cloud.py

```diff
@@ -220,3 +221,164 @@ def test_winrm_pinnned_version():
             winrm_pkg = pkg_resources.get_distribution("pywinrm")
             assert winrm_pkg.version >= '0.3.0'
     # fmt: on
+
+
+def test_ssh_gateway_arguments_default_alive_args():
+    arguments = ssh_gateway_arguments({"ssh_gateway": "host"})
+    assert "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL) in arguments
+    assert "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX) in arguments
+
+
+def test_ssh_gateway_arguments_alive_args():
+    arguments = ssh_gateway_arguments(
+        {
+            "ssh_gateway": "host",
+            "server_alive_interval": cloud.SERVER_ALIVE_INTERVAL + 1,
+            "server_alive_count_max": cloud.SERVER_ALIVE_COUNT_MAX * 2,
+        }
+    )
+    assert (
+        "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL + 1) in arguments
+    )
+    assert (
+        "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX * 2) in arguments
+    )
+
+
+@patch("salt.utils.cloud.socket", MagicMock())
+def test_wait_for_port_default_alive_args():
+    with patch(
+        "salt.utils.cloud._exec_ssh_cmd", MagicMock(return_value=0)
+    ) as exec_ssh_cmd:
+        cloud.wait_for_port(
+            "127.0.0.1", gateway={"ssh_gateway": "host", "ssh_gateway_user": "user"},
+        )
+        assert exec_ssh_cmd.call_count == 2
+        ssh_call = exec_ssh_cmd.call_args[0][0]
+        assert (
+            "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL) in ssh_call
+        )
+        assert (
+            "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX) in ssh_call
+        )
+
+
+@patch("salt.utils.cloud.socket", MagicMock())
+def test_wait_for_port_alive_args():
+    with patch(
+        "salt.utils.cloud._exec_ssh_cmd", MagicMock(return_value=0)
+    ) as exec_ssh_cmd:
+        cloud.wait_for_port(
+            "127.0.0.1",
+            server_alive_interval=cloud.SERVER_ALIVE_INTERVAL * 2,
+            server_alive_count_max=cloud.SERVER_ALIVE_COUNT_MAX + 1,
+            gateway={"ssh_gateway": "host", "ssh_gateway_user": "user"},
+        )
+        assert exec_ssh_cmd.call_count == 2
+        ssh_call = exec_ssh_cmd.call_args[0][0]
+        assert (
+            "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL * 2)
+            in ssh_call
+        )
```

Comment:
**suggestion (non-blocking)**: to avoid (potential) logic errors in tests, it's best to have a limited amount of logic in tests. Rather than tie this to `cloud.SERVER_ALIVE_INTERVAL`, just do something like

```
server_alive_interval = 42
expected_interval = " -oServerAliveInterval=42 "
server_alive_count_max = 13
expected_alive_count_max = " -oServerAliveCountCountMax=13 "

...

cloud.wait_for_port("127.0.0.1", server_alive_interval=server_alive_interval, server_alive_count_max=server_alive_count_max, ...)

...

assert expected_server_alive in ssh_call
assert expected_server_alive_count_max in ssh_call
```

Yeah, this is sort of anti-DRY, but it's also _very_ explicit about what the resulting values should be. In the tradeoff, I find that being explicit about the expected values is typically worth it.